src/components/common/GoodsItem/index.tsx

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import styled from '@emotion/styled';
+import Image from '@components/common/Image';
+import Ranking from './Ranking';
+
+const IMAGE_SIZE_RANKING = 160;
+const IMAGE_SIZE_GOODS_ITEM = 250;
+const IMAGE_RADIUS = 4;
+
+export interface GoodsItemProps {
+  imageSrc: string;
+  subtitle: string;
+  title: string;
+  amount: number;
+  rankingIndex?: number;
+}
+
+export default function GoodsItem({ imageSrc, subtitle, title, amount, rankingIndex, ...rest }: GoodsItemProps) {
+  const imageSize = rankingIndex ? IMAGE_SIZE_RANKING : IMAGE_SIZE_GOODS_ITEM;
+
+  const renderRanking = () => rankingIndex && <Ranking {...{ rankingIndex }} />;
+
```

Comment:
```suggestion
  const renderRanking = () => rankingIndex && <Ranking rankingIndex={rankingIndex} />;
```

지금처럼 spread operator 사용해서 전달하게 되면 `<Ranking {...{ rankingIndex, a: 1 }} />`처럼 컴포넌트에서 지원하지 않는 속성 (`a`)이 상위 컴포넌트에서 전달되더라도 정적 시점에 타입으로 검사되지 않아요.